app/src/main/java/com/duckduckgo/app/browser/webview/MaliciousSiteBlockerWebViewIntegration.kt

```diff
@@ -243,7 +270,7 @@ class RealMaliciousSiteBlockerWebViewIntegration @Inject constructor(
             } else {
                 Safe
             }
-            processedUrls.clear()
+            processedUrls[url] = isMalicious
```

Comment:
We have a previous check  `if (checkId == currentCheckId.get())` that validates this is related to the latest site trying to load. If we detect another site has started loading, we return safe to avoid showing the phishing page in the wrong context. However, we are assigning Safe to possible phishing pages here. I believe the right value here is `processedUrls[url] = it`.

Reply:
That makes a lot of sense, thanks!